### PR TITLE
HIVE-25795: [CVE-2021-44228] Update log4j2 version to 2.15.0

### PR DESCRIPTION
**Pull request by guptanikhil007**
### What changes were proposed in this pull request?

Update log4j2 version to 2.15.0

### Does this PR introduce _any_ user-facing change?

To avoid Apache Log4j RCE Zero day exploit in all versions < 2.15.0

### CODE REVIEW

**Comment by guptanikhil007** [comment]
@dengzhhu653 Closed previous pull request because build was stuck in pending to build state. (Somehow throttled)

Adding the conversation here:

bin/hive-config.sh changes
dengzhhu653:
maybe we do not need this after updating?

guptanikhil007:
Similar HBase fix:
https://github.com/apache/hbase/pull/3933/files

The leader of knownsec 404 team (ZoomEye & SeeBug) 'Heige' have also recommend to set log4j2.formatMsgNoLookups to true
Ref: https://www.cyberkendra.com/2021/12/worst-log4j-rce-zeroday-dropped-on.html

dengzhhu653:
Got it, thank you!

**Comment by dengzhhu653** [comment]
look good to me, +1

**Comment by aajisaka** [comment]
+1 (non-binding)

**Comment by brahmareddybattula** [comment]
+1 (non-binding)

**Comment by brahmareddybattula** [comment]
we need to backport to other branches also..branch-3.1, branc-2* also

File: bin/hive-config.sh [hunk 1]
```diff
@@ -68,3 +68,7 @@ export HIVE_AUX_JARS_PATH=$HIVE_AUX_JARS_PATH
 
 # Default to use 256MB 
 export HADOOP_HEAPSIZE=${HADOOP_HEAPSIZE:-256}
+
+# Disable the JNDI. This feature has critical REC vulnerability.
```
Comment (cnauroth) [inline bin/hive-config.sh:72]: Small nitpick on a typo: "RCE" instead of "REC".

**Review by cnauroth — CHANGES_REQUESTED** [review]
@guptanikhil007 , thank you for putting together this patch.  I entered one comment about a minor typo.  After correcting that, it looks good to me.

+1 (non-binding)

I think it would be good to patch branch-2 also for users still on the 2.x release line.  On branch-2, there is an additional occurrence of `<log4j2.version>2.6.2</log4j2.version>` in testutils/ptest2/pom.xml.

File: bin/hive-config.sh (hunk 1 above)
Comment (guptanikhil007) [reply bin/hive-config.sh]: Thanks for the review. 
Updated the comment

**Comment by dongjoon-hyun** [comment]
+1, LGTM.

**Comment by tasanuma** [comment]
FYI, the release vote for Log4j 2.15.1 has started, and it may be released soon.

**Review by sankarh — APPROVED** [review]

**Comment by brahmareddybattula** [comment]
@guptanikhil007 thanks for contributions ..are you planning raise PR for branch-2.X and branch-3.X also..?

**Comment by kevinverhoeven** [comment]
@guptanikhil007 thank you for this change, are you planning a fix for 2.x and 3.x? This fix is applied to 4.x which has not been released.

**Comment by brahmareddybattula** [comment]
> @guptanikhil007 thank you for this change, are you planning a fix for 2.x and 3.x? This fix is applied to 4.x which has not been released.

Looks PR is raised for branch-3. (#2869 ) . I will link this PR to jira so that it will be known.

**Pull request merged**